home/tests/test_views.py

```diff
@@ -0,0 +1,158 @@
+from urllib.parse import urlencode
+
+from django.contrib.auth import get_user_model
+from django.urls import reverse
+from rest_framework import status
+from rest_framework.test import APITestCase
+
+from .utils import create_page
+from home.models import ContentPageRating
+from home.serializers import ContentPageRatingSerializer, PageViewSerializer
+
+
+class PageRatingTestCase(APITestCase):
+    url = reverse("contentpagerating-list")
+
+    def test_page_rating_success(self):
+        user = get_user_model().objects.create_user("test")
+        self.client.force_authenticate(user)
+
+        page = create_page()
+
+        response = self.client.post(
+            self.url,
+            {
+                "page": page.id,
+                "helpful": False,
+                "comment": "lekker comment",
+                "data": {"contact_uuid": "123"},
+            },
+            format="json",
+        )
+
+        self.assertEquals(response.status_code, status.HTTP_201_CREATED)
+
+        response_data = response.json()
+        response_data.pop("timestamp")
+        rating = ContentPageRating.objects.last()
+        self.assertEquals(
+            response.json(),
+            {
+                "id": rating.id,
+                "helpful": False,
+                "comment": "lekker comment",
+                "data": {"contact_uuid": "123"},
+                "page": page.id,
+                "revision": page.get_latest_revision().id,
+            },
+        )
+
+    def test_page_rating_required_fields(self):
+        user = get_user_model().objects.create_user("test")
+        self.client.force_authenticate(user)
+
+        response = self.client.post(self.url, {}, format="json")
+
+        self.assertEquals(response.status_code, status.HTTP_400_BAD_REQUEST)
+        self.assertEquals(
+            response.json(),
+            {
+                "helpful": ["This field is required."],
+                "page": ["This field is required."],
+                "revision": ["This field is required."],
+            },
+        )
+
+    def test_page_rating_invalid_page(self):
+        user = get_user_model().objects.create_user("test")
+        self.client.force_authenticate(user)
+
+        response = self.client.post(self.url, {"page": 123}, format="json")
+
+        self.assertEquals(response.status_code, status.HTTP_400_BAD_REQUEST)
+        self.assertEquals(
+            response.json(),
+            {
+                "page": ["Page matching query does not exist."],
+            },
+        )
+
+    def test_get_list(self):
```

Comment:
This test and the one below would still pass if the view was incorrectly only returning 1 object. Perhaps you could add a 3rd object so 2 are returned, or make a separate test to confirm the querystring filtering is working?